content/api/vision.md

```diff
@@ -0,0 +1,72 @@
+  # Do robot stuff
+```
+
+## Marker properties
+The markers that are contained within the array have some useful properties:
+
+  - `is_wall_marker()` - returns whether or not the marker is a wall marker.
+  - `is_token_marker()` - returns whether or not the marker is a token marker.
+  - `id` - returns the id of the marker.
+  - `pixel_centre` - returns the pixel coordinates of the centre of the marker.
+  - `polar` - returns a `PolarCoord` object containing the position of the marker in the Polar coordinate system.
+  - `cartesian` - returns a `CartCoord` object containing the position of the marker in the Cartesian coordinate system.
+
+## Cartesian and Polar Coordinate Systems
+The markers' position can be represented using both the Polar and Cartesian coordinate system. The two objects `CartCoord` and `PolarCoord` hold the position information for both of these systems.
+
+{{% notice info %}}
+[Cartesian coordinates](https://en.wikipedia.org/wiki/Cartesian_coordinate_system) are ones that are defined using a unique set of values (typically x, y and z) to specify a point in space. [Polar coordinates](https://en.wikipedia.org/wiki/Polar_coordinate_system) are represented by a distance from a set point and an angle from a set direction.
+{{% /notice %}}
+
+{{% notice info %}}
+All distances in this section are measured in meters.
+{{% /notice %}}
+
+The `CartCoord` object has the following properties, where each coordinate is in Cartesian space:
+
+  - `x` - returns the x coordinate, defined as being positive in the right direction.
+  - `y` - returns the y coordinate, defined as being positive in the up direction.
+  - `z` - returns the z coordinate, defined as the distance from the camera to the marker.
+
+{{% notice info %}}
+For example, if x, y and z were 0, 0 and 1 respectively. This would correspond to a marker being exactly 1 metre in front of the camera, looking at the centre.
+{{% /notice %}}
+  
+The `PolarCoord` object has the following properties:
+ - `distance_meters` - returns the distance from the camera to the marker.
+ - `rot_x_rad` and `rot_x_deg` - returns the angle of rotation in radians/degrees about the x axis. The x axis runs horizontally in front of the camera. A positive angle would be one where the marker is leaning toward the camera.
+ - `rot_y_rad` and `rot_x_deg` - returns the angle of rotation in radians/degrees about the y axis. The y axis runs vertically in front of the camera. A positive angle would be one where the marker is rotated clockwise about the y axis looking from the top of the marker.
+
+## Marker IDs
+Each marker used in the game has a unique ID that corresponds to the marker type. You can use this within your code to determine what sort of marker you're looking at and make decisions based off of it.
+
+There are three main marker types in the game `WALL`, `TOKEN` and `TOKEN_ZONE_x` where `x` is the starting zone in the arena:
+  - `WALL`- Markers that are on the walls with IDs from 0 - 27
```

Comment:
Why is `WALL` here in backticks? Are we describing a constant which can be used in code? If so, we should be explicit about that. Otherwise we should remove the implication that it's code.

I also dislike that we're specifying the ranges at all -- it couples this page to the game much more than is needed, but misses out some game specifics. I'd much rather drop this and instead encourage use of either named constants in the code (for the type of the marker), or the methods like `is_wall_marker()` on the `Marker` instance.

Reply:
I'm not sure it's possible to fully decouple the pages from the game itself isn't especially possible. I think the best solution here however will be to create a sub-page off vision for the game-specific components of Vision

Reply:
Sure. I've realised that it's not so much the game specifics which I'm not enjoying here, as the inconsistency between whether we include game information and the level of detail we give.

I think it's ok for this page (without the need for a sub-page) to be coupled to the game, though I'd rather not expose the actual ids used since they should be an implementation detail on our side. I'd even go so far as to suggest that the rules also shouldn't include the ids (and instead maybe reference a constant).

I don't think we should remove the ids from the rules now that they're published, though I don't see any benefit to providing them in the docs.

I suggest that we:
 - remove the ids from here
 - clarify whether or not these are constants
 - add some documentation about the markers which are on the columns